src/main/webapp/user_profile_script.js

```diff
@@ -52,10 +104,105 @@ function closeModal() {
     modal.style.display = "none";
 }
 
+function closeEditModal() {
+    var modal = document.getElementById("editTaskModal");
+    modal.style.display = "none";
+}
+
 // If the user clicks outside of the modal, closes the modal directly
 window.onclick = function(event) {
     var modal = document.getElementById("createTaskModal");
     if (event.target == modal) {
         modal.style.display = "none";
     }
+    var editModal = document.getElementById("editTaskModal");
+    if (event.target == editModal) {
+        editModal.style.display = "none";
+    }
+}
+
+async function displayNeedHelpTasks() {
+    const queryURL = "mytasks?keyword=Owner";
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const taskResponse = await response.json();
+    const needHelpTable = document.getElementById("need-help");
+    needHelpTable.innerHTML = "";
+    var headerRow = document.createElement("tr");
+    var headers = ["Task Overview", "Helper", "Status", "Edit", "Delete"];
+    for (var index = 0; index < headers.length; index++) {
+        var th = document.createElement("th");
+        th.appendChild(document.createTextNode(headers[index]));
+        headerRow.appendChild(th);
+    }
+    needHelpTable.appendChild(headerRow);
+    for (var index = 0; index < taskResponse.length; index++) {
+        var tr = document.createElement("tr");
+        var task = taskResponse[index];
+        var data = [task.detail, task.helper, task.status];
+        for (var i = 0; i < data.length; i++) {
+            var td = document.createElement("td");
+            td.appendChild(document.createTextNode(data[i]));
+            tr.appendChild(td);
+        }
+        const keyStringCopy = task.keyString.slice();
+        var editTd = document.createElement("td");
+        var editBtn = document.createElement("button");
+        editBtn.className = "edit-task";
+        editBtn.addEventListener("click", function () { editTask(keyStringCopy) });
+        editBtn.innerHTML = (task.status === "OPEN") ? '<i class="fa fa-edit"></i>':'<i class="fa fa-ban"></i>';
+        editTd.appendChild(editBtn);
+        var deleteTd = document.createElement("td");
+        var deleteBtn = document.createElement("button");
+        deleteBtn.className = "delete-task";
+        deleteBtn.addEventListener("click", function () { deleteTask(keyStringCopy) });
+        deleteBtn.innerHTML = (task.status === "OPEN") ? '<i class="fa fa-trash-o"></i>':'<i class="fa fa-ban"></i>';
+        deleteTd.appendChild(deleteBtn);
+        tr.appendChild(editTd);
+        tr.appendChild(deleteTd);
+        needHelpTable.appendChild(tr);
```

Comment:
Something to consider: you can make a template row in your HTML file that's hidden, then this function can clone it, replace all the relevant fields with Task-specific information, then append it to the table. That should simplify and shorten this function a bit.

Reply:
Oh the suggestion is really helpful. It can make the two functions `displayNeedHelpTasks()` and `displayOfferHelpTasks()` much more concise. I will try this way in a new PR.

Reply:
Sure, sounds good.